authentication/auth.go

```diff
@@ -151,18 +151,45 @@ func generateClaims(postValues url.Values) (claims eqClaims) {
 		RuRef:          postValues.Get("ru_ref"),
 		RuName:         postValues.Get("ru_name"),
 		RefPStartDate:  postValues.Get("ref_p_start_date"),
-		RefPEndDate:    postValues.Get("ref_p_end_date"),
 		ReturnBy:       postValues.Get("return_by"),
-		TradAs:         postValues.Get("trad_as"),
-		EmploymentDate: postValues.Get("employment_date"),
-		RegionCode:     postValues.Get("region_code"),
-		LanguageCode:   postValues.Get("language_code"),
-		TxID:           uuid.NewV4().String(),
-		Roles:          postValues.Get("roles"),
-		VariantFlags: variantFlags{
+	}
+
+	// Add optional values 
+	if postValues.Get("ref_p_end_date") != "" {
+		claims.RefPEndDate = postValues.Get("ref_p_end_date")
+	}
+
+	if postValues.Get("trad_as") != "" {
+		claims.TradAs = postValues.Get("trad_as")
+	}	
+
+	if postValues.Get("employment_date") != "" {
+		claims.EmploymentDate = postValues.Get("employment_date")
+	}	
+
+	if postValues.Get("region_code") != "" {
+		claims.RegionCode = postValues.Get("region_code")
+	}	
+
+	if postValues.Get("language_code") != "" {
+		claims.LanguageCode = postValues.Get("language_code")
+	}
+
+	if postValues.Get("roles") != "" {
+		claims.Roles = postValues.Get("roles")
+	}
```

Comment:
My first reaction is it feels like there should be a more idiomatic way of doing this. Having said that I think it would be a lot clearer if the go app was a JSON API and the form was dealt with client side, so maybe we should prioritise that.

Reply:
As discussed, we'll do this outside this PR potentially